app/models/manageiq/providers/vmware/infra_manager/orchestration_template.rb

```diff
@@ -47,6 +47,12 @@ def deployment_spec(opts)
       target[SPEC_KEY_MAPPING[r]] = resource.ems_ref if resource.present?
     end
 
+    if target["folder_id"].blank? && opts[:datacenter_id].present?
+      dc = Datacenter.find_by(:id => opts[:datacenter_id])
+      vm_folder = dc.folders.find { |f| f.name == 'vm' } if dc
+      target["folder_id"] = vm_folder.ems_ref if vm_folder
```

Comment:
Why are we assigning an ems_ref to an _id field?

Reply:
Or is `target["folder_id"]` what is being sent to the vsphere api?

Reply:
That is right. Vshpere API expects a hash with keys in string.